src/registers/model_specific.rs

```diff
@@ -112,6 +121,26 @@ bitflags! {
     }
 }
 
+bitflags! {
+    /// Configurable (RW) flags stored in IA32_U_CET
+    pub struct UCetFlags: u64 {
+        /// Enable shadow stack (SH_STK_EN)
+        const SS_ENABLE = 1 << 0;
+        /// Enable WRSS{D,Q}W instructions (WR_SHTK_EN)
+        const SS_WRITE_ENABLE = 1 << 1;
+        /// Enable indirect branch tracking (ENDBR_EN)
+        const IBT_ENABLE = 1 << 2;
+        /// Enable legacy treatment for indirect branch tracking (LEG_IW_EN)
+        const IBT_LEGACY_ENABLE = 1 << 3;
+        /// Enable no-track opcode prefix for indirect branch tracking (NO_TRACK_EN)
+        const IBT_NO_TRACK_ENABLE = 1 << 4;
+        /// Disable suppression of CET on legacy compatibility (SUPPRESS_DIS)
+        const IBT_LEGACY_SUPPRESS_ENABLE = 1 << 5;
+        /// Enable suppression of indirect branch tracking (SUPPRESS)
+        const IBT_SUPPRESS_ENABLE = 1 << 10;
+    }
```

Comment:
The Intel SDM also lists a `TRACKER` flag. This probably isn't intended to be read or written directly, but we could still add it just for completeness sake.

Reply:
There is also a bitmap after that. Can `bitfield` include a range of bits?

Reply:
> There is also a bitmap after that. Can `bit_field` include a range of bits?

No, it can't. See this comment https://github.com/rust-osdev/x86_64/pull/349#discussion_r829828934.

Reply:
Perhaps I'll add a new flag called `IBT_TRACKED`:

```
// Is IBT waiting for a branch to return? (read-only, TRACKER)
const IBT_TRACKED = 1 << 11;
```

There is also the address of a legacy code page bitmap. It's bitmap where where `1` represents that 4k page is non-CET treated legacy code (i.e. do not track code page). For that I would have separate read/write because it is something that you setup when you e.g. load your program. See 18.3.6.1 in SDM Vol 1 for more information.

So perhaps something like `fn write_legacy_bitmap_addr(addr: &Page)` and `fn write_legacy_bitmap_addr(addr: &Page)`?

Reply:
> Perhaps I'll add a new flag called `IBT_TRACKED`:
> 
> ```
> // Is IBT waiting for a branch to return? (read-only, TRACKER)
> const IBT_TRACKED = 1 << 11;
> ```

+1

> There is also the address of a legacy code page bitmap. It's bitmap where where `1` represents that 4k page is non-CET treated legacy code (i.e. do not track code page). For that I would have separate read/write because it is something that you setup when you e.g. load your program. See 18.3.6.1 in SDM Vol 1 for more information.
> 
> So perhaps something like `fn read_legacy_bitmap_addr() -> &Page` and `fn write_legacy_bitmap_addr(addr: &Page)`?
> 

See https://github.com/rust-osdev/x86_64/pull/349#discussion_r829828934